tokio/src/sync/semaphore.rs

```diff
@@ -76,6 +76,51 @@ use std::sync::Arc;
 /// }
 /// ```
 ///
+/// ## Limit the number of outgoing requests being sent at the same time
+///
+/// In some scenarios, it might be required to limit the number of outgoing
+/// requests being sent in parallel. This could be due to limits of a consumed
+/// API or the network resources of the system the application is running on.
+///
+/// This example uses an `Arc<Semaphore>` with 10 permits.
+/// Each task spawned is given a reference to the semaphore by cloning the `Arc<Semaphore>`.
+/// Before a task sends a request, it must acquire a permit from the semaphore by calling [`Semaphore::acquire`].
+/// This ensures that at most 10 requests are sent in parallel at any given time.
+/// After a task has sent a request, it drops the permit to allow other tasks to send requests.
```

Comment:
This markdown has a newline after each sentence, but that's not how it will be rendered in the browser. Screenshot:

![image](https://github.com/tokio-rs/tokio/assets/928074/f894b46c-1041-4883-9881-8f5a7f1877c9)

I would prefer if you reflow these sentences so that each line goes as close as it can to a maximum of 80 characters per line, like you did in the previous paragraph.

Or if you intended for this to be a list, then add bullet points to make it into an actual list.